src/cpp/server_core/include/server_core/ServerDatabase.hpp

```diff
@@ -16,23 +16,21 @@
 #ifndef SERVER_CORE_SERVER_DATABASE_HPP
 #define SERVER_CORE_SERVER_DATABASE_HPP
 
+#include <boost/optional.hpp>
+
 #include <core/Database.hpp>
+
 #include <shared_core/Error.hpp>
+#include <shared_core/system/User.hpp>
 
 namespace rstudio {
-
-namespace core {
```

Comment:
Is it safe to remove the `#include <shared_core/Error.hpp>` rather than the forward declaration? It's not too important, since we'd just end up including `Error.hpp` pretty much everywhere, but I definitely have a preference for forcing each file to explicitly include all the files it needs.

Reply:
We already had a full declaration in this header file, so I just removed the forward decl (it was already declared).